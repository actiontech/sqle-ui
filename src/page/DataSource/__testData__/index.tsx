export const dataSourceList = [
  {
    instance_name: 'db1',
    db_host: '20.20.20.2',
    db_port: '3306',
    db_user: 'root',
    desc: '',
    workflow_template_name: 'default',
  },
];

export const dataSourceInstance = {
  instance_name: 'db1',
  db_host: '20.20.20.2',
  db_port: '3306',
  db_user: 'root',
  db_type: 'mysql',
  desc: '',
  workflow_template_name: 'workflow-template-name-1',
};

export const dataSourceMetas = [
  {
    db_type: 'mysql',
    params: [
      { description: '字段a', name: 'a', type: 'string', value: '123' },
      { description: '字段b', name: 'b', type: 'int', value: '123' },
      { description: '字段c', name: 'c', type: 'bool', value: 'true' },
    ],
  },
  {
    db_type: 'oracle',
    params: [
      { description: '字段e', name: 'e', type: 'string', value: '2222' },
      { description: '字段f', name: 'f', type: 'int', value: '333' },
      { description: '字段mm', name: 'mm', type: 'bool', value: 'false' },
    ],
  },
];
