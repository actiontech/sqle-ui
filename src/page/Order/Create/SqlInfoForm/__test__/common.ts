import instance from '../../../../../api/instance';
import { resolveThreeSecond } from '../../../../../testUtils/mockRequest';

export const mockGetInstance = () => {
  const spy = jest.spyOn(instance, 'getInstanceV2');
  spy.mockImplementation(() =>
    resolveThreeSecond({
      instance_name: 'db1',
      db_host: '20.20.20.2',
      db_port: '3306',
      db_user: 'root',
      db_type: 'mysql',
      desc: '',
      workflow_template_name: 'workflow-template-name-1',
      rule_template: {
        name: 'not_submit_test_rule33',
        is_global_rule_template: true,
      },
    })
  );

  return spy;
};
