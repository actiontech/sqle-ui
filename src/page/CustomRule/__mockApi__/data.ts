import { ICustomRuleResV1 } from '../../../api/common';
import { CustomRuleResV1LevelEnum } from '../../../api/common.enum';

export const customRules: ICustomRuleResV1[] = [
  {
    db_type: 'mysql',
    annotation: 'desc desc desc',
    level: CustomRuleResV1LevelEnum.notice,
    rule_id: '1',
    desc: 'name1',
    rule_script: 'w+',
    type: 'rule_type1',
  },
  {
    db_type: 'oracle',
    annotation: 'desc desc desc',
    level: CustomRuleResV1LevelEnum.error,
    rule_id: '2',
    desc: 'name2',
    rule_script: 'w+',
    type: 'rule_type2',
  },
];
