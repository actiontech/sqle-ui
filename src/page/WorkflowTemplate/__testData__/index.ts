import { WorkflowTemplateDetailResV1AllowSubmitWhenLessAuditLevelEnum } from "../../../api/common.enum";

export const workflowData = {
  allow_submit_when_less_audit_level: WorkflowTemplateDetailResV1AllowSubmitWhenLessAuditLevelEnum.warn,
  workflow_template_name: 'db1',
  desc: '123',
  workflow_step_template_list: [
    {
      number: 1,
      type: 'sql_review',
      desc: '1',
      assignee_user_name_list: ['admin'],
    },
    {
      number: 2,
      type: 'sql_review',
      desc: '2',
      assignee_user_name_list: ['admin'],
      approved_by_authorized: true,
    },
    {
      number: 3,
      type: 'sql_execute',
      desc: '3',
      assignee_user_name_list: ['admin'],
    },
  ],
  instance_name_list: ['db1'],
};

export const workflowData2 = {
  allow_submit_when_less_audit_level: WorkflowTemplateDetailResV1AllowSubmitWhenLessAuditLevelEnum.error,
  workflow_template_name: 'db1',
  desc: '123',
  workflow_step_template_list: [
    {
      number: 1,
      type: 'sql_review',
      desc: '1',
      assignee_user_name_list: ['admin'],
    },
    {
      number: 2,
      type: 'sql_review',
      desc: '2',
      assignee_user_name_list: [],
      approved_by_authorized: true,
    },
    {
      number: 3,
      type: 'sql_execute',
      desc: '3',
      assignee_user_name_list: [],
      execute_by_authorized: true,
    },
  ],
  instance_name_list: ['db1'],
};
