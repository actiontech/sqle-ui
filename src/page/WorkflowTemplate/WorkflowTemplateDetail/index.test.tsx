import { cleanup, screen, act } from '@testing-library/react';
import WorkflowTemplateDetail from '.';
import workflow from '../../../api/workflow';
import { renderWithThemeAndRouter } from '../../../testUtils/customRender';
import { resolveThreeSecond } from '../../../testUtils/mockRequest';
import { workflowData, workflowData2 } from '../__testData__';
import { useParams } from 'react-router-dom';
import { SystemRole } from '../../../data/common';
import { useSelector } from 'react-redux';
import { mockBindProjects } from '../../../hooks/useCurrentUser/index.test.data';

jest.mock('react-redux', () => {
  return {
    ...jest.requireActual('react-redux'),
    useSelector: jest.fn(),
  };
});

jest.mock('react-router', () => {
  return {
    ...jest.requireActual('react-router'),
    useParams: jest.fn(),
  };
});

const projectName = 'default';
describe('WorkflowTemplate/WorkflowTemplateDetail', () => {
  const useParamsMock: jest.Mock = useParams as jest.Mock;
  let getWorkflowTemplateDetail!: jest.SpyInstance;

  const mockGetWorkflowTemplateDetail = () => {
    const spy = jest.spyOn(workflow, 'getWorkflowTemplateV1');
    spy.mockImplementation(() => resolveThreeSecond(workflowData));
    return spy;
  };

  beforeEach(() => {
    jest.useFakeTimers();
    useParamsMock.mockReturnValue({ workflowName: 'default', projectName });
    getWorkflowTemplateDetail = mockGetWorkflowTemplateDetail();

    (useSelector as jest.Mock).mockImplementation((e) =>
      e({
        user: { role: SystemRole.admin, bindProjects: mockBindProjects },
        projectManage: { archived: false },
      })
    );
  });

  afterEach(() => {
    jest.useRealTimers();
  });

  test('should match snapshot', async () => {
    const { container } = renderWithThemeAndRouter(<WorkflowTemplateDetail />);
    expect(container).toMatchSnapshot();
    expect(getWorkflowTemplateDetail).toBeCalledWith({
      project_name: projectName,
    });
    await act(async () => jest.advanceTimersByTime(3000));

    expect(container).toMatchSnapshot();
  });

  test('should hide update template feature when not currently a project manager or admin', async () => {
    (useSelector as jest.Mock).mockImplementation((e) =>
      e({
        user: {
          role: SystemRole.admin,
          bindProjects: [{ projectName: 'test', isManager: false }],
        },
        projectManage: { archived: false },
      })
    );

    renderWithThemeAndRouter(<WorkflowTemplateDetail />);
    await act(async () => jest.advanceTimersByTime(3000));

    expect(
      screen.getByText('workflowTemplate.detail.updateTemplate')
    ).toBeInTheDocument();

    cleanup();
    jest.clearAllMocks();

    (useSelector as jest.Mock).mockImplementation((e) =>
      e({
        user: {
          role: '',
          bindProjects: mockBindProjects,
        },
        projectManage: { archived: false },
      })
    );

    renderWithThemeAndRouter(<WorkflowTemplateDetail />);
    await act(async () => jest.advanceTimersByTime(3000));

    expect(
      screen.getByText('workflowTemplate.detail.updateTemplate')
    ).toBeInTheDocument();

    cleanup();
    jest.clearAllMocks();

    (useSelector as jest.Mock).mockImplementation((e) =>
      e({
        user: {
          role: '',
          bindProjects: [{ projectName: 'default', isManager: false }],
        },
        projectManage: { archived: false },
      })
    );
    renderWithThemeAndRouter(<WorkflowTemplateDetail />);
    await act(async () => jest.advanceTimersByTime(3000));

    expect(
      screen.queryByText('workflowTemplate.detail.updateTemplate')
    ).not.toBeInTheDocument();
  });

  test('should render match text when choosing to match All privileged users', async () => {
    getWorkflowTemplateDetail.mockImplementation(() =>
      resolveThreeSecond(workflowData2)
    );
    const { container } = renderWithThemeAndRouter(<WorkflowTemplateDetail />);
    await act(async () => jest.advanceTimersByTime(3000));

    expect(container).toMatchSnapshot();
    expect(
      screen.getByText(
        'workflowTemplate.form.label.reviewUser : workflowTemplate.progressConfig.review.reviewUserType.matchAudit'
      )
    ).toBeInTheDocument();
    expect(
      screen.getByText(
        'workflowTemplate.form.label.execUser : workflowTemplate.progressConfig.exec.executeUserType.matchExecute'
      )
    ).toBeInTheDocument();
  });

  test('should display the highest audit level allowed to create a order', async () => {
    getWorkflowTemplateDetail.mockImplementation(() =>
      resolveThreeSecond(workflowData)
    );

    renderWithThemeAndRouter(<WorkflowTemplateDetail />);
    await act(async () => jest.advanceTimersByTime(3000));

    expect(
      screen.getByText('workflowTemplate.auditLevel.warn')
    ).toBeInTheDocument();
  });

  test('should hide the Create, Delete, Edit feature when project is archived', async () => {
    (useSelector as jest.Mock).mockImplementation((e) =>
      e({
        user: {
          role: SystemRole.admin,
          bindProjects: [{ projectName, isManager: true }],
        },
        projectManage: { archived: true },
      })
    );

    renderWithThemeAndRouter(<WorkflowTemplateDetail />);
    await act(async () => jest.advanceTimersByTime(3000));

    expect(
      screen.queryByText('workflowTemplate.detail.updateTemplate')
    ).not.toBeInTheDocument();
  });
});
