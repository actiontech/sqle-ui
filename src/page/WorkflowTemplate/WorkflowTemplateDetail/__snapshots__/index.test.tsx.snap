// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowTemplate/WorkflowTemplateDetail should match snapshot 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          workflowTemplate.detail.title.wrapper
        </div>
        <div
          class="ant-card-extra"
        >
          <a
            href="/project/default/progress/update/undefined"
          >
            <button
              class="ant-btn ant-btn-primary"
              type="button"
            >
              <span>
                workflowTemplate.detail.updateTemplate
              </span>
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <div
        class="ant-row"
      >
        <div
          class="ant-col ant-col-24"
          style="margin-bottom: 6px;"
        >
          <div
            class="ant-space ant-space-horizontal ant-space-align-baseline"
          >
            <div
              class="ant-space-item"
              style="margin-right: 16px;"
            >
              <h5
                class="ant-typography"
              >
                workflowTemplate.form.label.allowSubmitWhenLessAuditLevel
              </h5>
            </div>
            <div
              class="ant-space-item"
            >
              <span
                class="ant-typography"
              >
                
              </span>
            </div>
          </div>
        </div>
        <div
          class="ant-col ant-col-24"
        >
          <h5
            class="ant-typography"
          >
            workflowTemplate.detail.title.step
          </h5>
          <div
            class="ant-steps ant-steps-vertical"
          >
            <div
              class="ant-steps-item ant-steps-item-process ant-steps-item-active"
              style="margin-bottom: 10px;"
            >
              <div
                class="ant-steps-item-container"
              >
                <div
                  class="ant-steps-item-tail"
                />
                <div
                  class="ant-steps-item-icon"
                >
                  <span
                    class="ant-steps-icon"
                  >
                    1
                  </span>
                </div>
                <div
                  class="ant-steps-item-content"
                >
                  <div
                    class="ant-steps-item-title"
                  >
                    workflowTemplate.progressConfig.createStep.title
                  </div>
                  <div
                    class="ant-steps-item-description"
                  >
                    workflowTemplate.progressConfig.createStep.desc
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-steps-item ant-steps-item-process"
            >
              <div
                class="ant-steps-item-container"
              >
                <div
                  class="ant-steps-item-tail"
                />
                <div
                  class="ant-steps-item-icon"
                >
                  <span
                    class="ant-steps-icon"
                  >
                    2
                  </span>
                </div>
                <div
                  class="ant-steps-item-content"
                >
                  <div
                    class="ant-steps-item-title"
                  >
                    workflowTemplate.progressConfig.exec.title
                    <div
                      class="ant-space ant-space-horizontal ant-space-align-center"
                    >
                      <div
                        class="ant-space-item"
                      >
                        <span
                          aria-label="info-circle"
                          class="anticon anticon-info-circle text-orange"
                          role="img"
                          style="font-size: 14px; margin-left: 6px;"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="info-circle"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                            />
                            <path
                              d="M464 336a48 48 0 1096 0 48 48 0 10-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
                            />
                          </svg>
                        </span>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-steps-item-description"
                  >
                    <div
                      class="ant-space ant-space-vertical full-width-element"
                    >
                      <div
                        class="ant-space-item"
                        style="margin-bottom: 0px;"
                      >
                        <span>
                          workflowTemplate.form.label.execUser
                           : 
                          --
                        </span>
                      </div>
                      <div
                        class="ant-space-item"
                      >
                        <span>
                          <span
                            class="text-black"
                          >
                            workflowTemplate.form.label.reviewDesc
                             : 
                          </span>
                          
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WorkflowTemplate/WorkflowTemplateDetail should match snapshot 2`] = `
<div>
  <div
    class="ant-card ant-card-bordered"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          workflowTemplate.detail.title.wrapper
        </div>
        <div
          class="ant-card-extra"
        >
          <a
            href="/project/default/progress/update/db1"
          >
            <button
              class="ant-btn ant-btn-primary"
              type="button"
            >
              <span>
                workflowTemplate.detail.updateTemplate
              </span>
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <div
        class="ant-row"
      >
        <div
          class="ant-col ant-col-24"
          style="margin-bottom: 6px;"
        >
          <div
            class="ant-space ant-space-horizontal ant-space-align-baseline"
          >
            <div
              class="ant-space-item"
              style="margin-right: 16px;"
            >
              <h5
                class="ant-typography"
              >
                workflowTemplate.form.label.allowSubmitWhenLessAuditLevel
              </h5>
            </div>
            <div
              class="ant-space-item"
            >
              <span
                class="ant-typography"
              >
                workflowTemplate.auditLevel.warn
              </span>
            </div>
          </div>
        </div>
        <div
          class="ant-col ant-col-24"
        >
          <h5
            class="ant-typography"
          >
            workflowTemplate.detail.title.step
          </h5>
          <div
            class="ant-steps ant-steps-vertical"
          >
            <div
              class="ant-steps-item ant-steps-item-process ant-steps-item-active"
              style="margin-bottom: 10px;"
            >
              <div
                class="ant-steps-item-container"
              >
                <div
                  class="ant-steps-item-tail"
                />
                <div
                  class="ant-steps-item-icon"
                >
                  <span
                    class="ant-steps-icon"
                  >
                    1
                  </span>
                </div>
                <div
                  class="ant-steps-item-content"
                >
                  <div
                    class="ant-steps-item-title"
                  >
                    workflowTemplate.progressConfig.createStep.title
                  </div>
                  <div
                    class="ant-steps-item-description"
                  >
                    workflowTemplate.progressConfig.createStep.desc
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-steps-item ant-steps-item-process"
              style="margin-bottom: 10px;"
            >
              <div
                class="ant-steps-item-container"
              >
                <div
                  class="ant-steps-item-tail"
                />
                <div
                  class="ant-steps-item-icon"
                >
                  <span
                    class="ant-steps-icon"
                  >
                    2
                  </span>
                </div>
                <div
                  class="ant-steps-item-content"
                >
                  <div
                    class="ant-steps-item-title"
                  >
                    workflowTemplate.progressConfig.review.title
                    <div
                      class="ant-space ant-space-horizontal ant-space-align-center"
                    >
                      <div
                        class="ant-space-item"
                      >
                        <span
                          aria-label="info-circle"
                          class="anticon anticon-info-circle text-orange"
                          role="img"
                          style="font-size: 14px; margin-left: 6px;"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="info-circle"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                            />
                            <path
                              d="M464 336a48 48 0 1096 0 48 48 0 10-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
                            />
                          </svg>
                        </span>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-steps-item-description"
                  >
                    <div
                      class="ant-space ant-space-vertical full-width-element"
                    >
                      <div
                        class="ant-space-item"
                        style="margin-bottom: 0px;"
                      >
                        <span>
                          workflowTemplate.form.label.reviewUser
                           : 
                          admin
                        </span>
                      </div>
                      <div
                        class="ant-space-item"
                      >
                        <span>
                          <span
                            class="text-black"
                          >
                            workflowTemplate.form.label.reviewDesc
                             : 
                          </span>
                          1
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-steps-item ant-steps-item-process"
              style="margin-bottom: 10px;"
            >
              <div
                class="ant-steps-item-container"
              >
                <div
                  class="ant-steps-item-tail"
                />
                <div
                  class="ant-steps-item-icon"
                >
                  <span
                    class="ant-steps-icon"
                  >
                    3
                  </span>
                </div>
                <div
                  class="ant-steps-item-content"
                >
                  <div
                    class="ant-steps-item-title"
                  >
                    workflowTemplate.progressConfig.review.title
                    <div
                      class="ant-space ant-space-horizontal ant-space-align-center"
                    >
                      <div
                        class="ant-space-item"
                      >
                        <span
                          aria-label="info-circle"
                          class="anticon anticon-info-circle text-orange"
                          role="img"
                          style="font-size: 14px; margin-left: 6px;"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="info-circle"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                            />
                            <path
                              d="M464 336a48 48 0 1096 0 48 48 0 10-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
                            />
                          </svg>
                        </span>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-steps-item-description"
                  >
                    <div
                      class="ant-space ant-space-vertical full-width-element"
                    >
                      <div
                        class="ant-space-item"
                        style="margin-bottom: 0px;"
                      >
                        <span>
                          workflowTemplate.form.label.reviewUser
                           : 
                          admin
                        </span>
                      </div>
                      <div
                        class="ant-space-item"
                      >
                        <span>
                          <span
                            class="text-black"
                          >
                            workflowTemplate.form.label.reviewDesc
                             : 
                          </span>
                          2
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-steps-item ant-steps-item-process"
            >
              <div
                class="ant-steps-item-container"
              >
                <div
                  class="ant-steps-item-tail"
                />
                <div
                  class="ant-steps-item-icon"
                >
                  <span
                    class="ant-steps-icon"
                  >
                    4
                  </span>
                </div>
                <div
                  class="ant-steps-item-content"
                >
                  <div
                    class="ant-steps-item-title"
                  >
                    workflowTemplate.progressConfig.exec.title
                    <div
                      class="ant-space ant-space-horizontal ant-space-align-center"
                    >
                      <div
                        class="ant-space-item"
                      >
                        <span
                          aria-label="info-circle"
                          class="anticon anticon-info-circle text-orange"
                          role="img"
                          style="font-size: 14px; margin-left: 6px;"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="info-circle"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                            />
                            <path
                              d="M464 336a48 48 0 1096 0 48 48 0 10-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
                            />
                          </svg>
                        </span>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-steps-item-description"
                  >
                    <div
                      class="ant-space ant-space-vertical full-width-element"
                    >
                      <div
                        class="ant-space-item"
                        style="margin-bottom: 0px;"
                      >
                        <span>
                          workflowTemplate.form.label.execUser
                           : 
                          admin
                        </span>
                      </div>
                      <div
                        class="ant-space-item"
                      >
                        <span>
                          <span
                            class="text-black"
                          >
                            workflowTemplate.form.label.reviewDesc
                             : 
                          </span>
                          3
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WorkflowTemplate/WorkflowTemplateDetail should render match text when choosing to match All privileged users 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          workflowTemplate.detail.title.wrapper
        </div>
        <div
          class="ant-card-extra"
        >
          <a
            href="/project/default/progress/update/db1"
          >
            <button
              class="ant-btn ant-btn-primary"
              type="button"
            >
              <span>
                workflowTemplate.detail.updateTemplate
              </span>
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <div
        class="ant-row"
      >
        <div
          class="ant-col ant-col-24"
          style="margin-bottom: 6px;"
        >
          <div
            class="ant-space ant-space-horizontal ant-space-align-baseline"
          >
            <div
              class="ant-space-item"
              style="margin-right: 16px;"
            >
              <h5
                class="ant-typography"
              >
                workflowTemplate.form.label.allowSubmitWhenLessAuditLevel
              </h5>
            </div>
            <div
              class="ant-space-item"
            >
              <span
                class="ant-typography"
              >
                workflowTemplate.auditLevel.error
              </span>
            </div>
          </div>
        </div>
        <div
          class="ant-col ant-col-24"
        >
          <h5
            class="ant-typography"
          >
            workflowTemplate.detail.title.step
          </h5>
          <div
            class="ant-steps ant-steps-vertical"
          >
            <div
              class="ant-steps-item ant-steps-item-process ant-steps-item-active"
              style="margin-bottom: 10px;"
            >
              <div
                class="ant-steps-item-container"
              >
                <div
                  class="ant-steps-item-tail"
                />
                <div
                  class="ant-steps-item-icon"
                >
                  <span
                    class="ant-steps-icon"
                  >
                    1
                  </span>
                </div>
                <div
                  class="ant-steps-item-content"
                >
                  <div
                    class="ant-steps-item-title"
                  >
                    workflowTemplate.progressConfig.createStep.title
                  </div>
                  <div
                    class="ant-steps-item-description"
                  >
                    workflowTemplate.progressConfig.createStep.desc
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-steps-item ant-steps-item-process"
              style="margin-bottom: 10px;"
            >
              <div
                class="ant-steps-item-container"
              >
                <div
                  class="ant-steps-item-tail"
                />
                <div
                  class="ant-steps-item-icon"
                >
                  <span
                    class="ant-steps-icon"
                  >
                    2
                  </span>
                </div>
                <div
                  class="ant-steps-item-content"
                >
                  <div
                    class="ant-steps-item-title"
                  >
                    workflowTemplate.progressConfig.review.title
                    <div
                      class="ant-space ant-space-horizontal ant-space-align-center"
                    >
                      <div
                        class="ant-space-item"
                      >
                        <span
                          aria-label="info-circle"
                          class="anticon anticon-info-circle text-orange"
                          role="img"
                          style="font-size: 14px; margin-left: 6px;"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="info-circle"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                            />
                            <path
                              d="M464 336a48 48 0 1096 0 48 48 0 10-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
                            />
                          </svg>
                        </span>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-steps-item-description"
                  >
                    <div
                      class="ant-space ant-space-vertical full-width-element"
                    >
                      <div
                        class="ant-space-item"
                        style="margin-bottom: 0px;"
                      >
                        <span>
                          workflowTemplate.form.label.reviewUser
                           : 
                          admin
                        </span>
                      </div>
                      <div
                        class="ant-space-item"
                      >
                        <span>
                          <span
                            class="text-black"
                          >
                            workflowTemplate.form.label.reviewDesc
                             : 
                          </span>
                          1
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-steps-item ant-steps-item-process"
              style="margin-bottom: 10px;"
            >
              <div
                class="ant-steps-item-container"
              >
                <div
                  class="ant-steps-item-tail"
                />
                <div
                  class="ant-steps-item-icon"
                >
                  <span
                    class="ant-steps-icon"
                  >
                    3
                  </span>
                </div>
                <div
                  class="ant-steps-item-content"
                >
                  <div
                    class="ant-steps-item-title"
                  >
                    workflowTemplate.progressConfig.review.title
                    <div
                      class="ant-space ant-space-horizontal ant-space-align-center"
                    >
                      <div
                        class="ant-space-item"
                      >
                        <span
                          aria-label="info-circle"
                          class="anticon anticon-info-circle text-orange"
                          role="img"
                          style="font-size: 14px; margin-left: 6px;"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="info-circle"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                            />
                            <path
                              d="M464 336a48 48 0 1096 0 48 48 0 10-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
                            />
                          </svg>
                        </span>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-steps-item-description"
                  >
                    <div
                      class="ant-space ant-space-vertical full-width-element"
                    >
                      <div
                        class="ant-space-item"
                        style="margin-bottom: 0px;"
                      >
                        <span>
                          workflowTemplate.form.label.reviewUser
                           : 
                          workflowTemplate.progressConfig.review.reviewUserType.matchAudit
                        </span>
                      </div>
                      <div
                        class="ant-space-item"
                      >
                        <span>
                          <span
                            class="text-black"
                          >
                            workflowTemplate.form.label.reviewDesc
                             : 
                          </span>
                          2
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-steps-item ant-steps-item-process"
            >
              <div
                class="ant-steps-item-container"
              >
                <div
                  class="ant-steps-item-tail"
                />
                <div
                  class="ant-steps-item-icon"
                >
                  <span
                    class="ant-steps-icon"
                  >
                    4
                  </span>
                </div>
                <div
                  class="ant-steps-item-content"
                >
                  <div
                    class="ant-steps-item-title"
                  >
                    workflowTemplate.progressConfig.exec.title
                    <div
                      class="ant-space ant-space-horizontal ant-space-align-center"
                    >
                      <div
                        class="ant-space-item"
                      >
                        <span
                          aria-label="info-circle"
                          class="anticon anticon-info-circle text-orange"
                          role="img"
                          style="font-size: 14px; margin-left: 6px;"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="info-circle"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                            />
                            <path
                              d="M464 336a48 48 0 1096 0 48 48 0 10-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
                            />
                          </svg>
                        </span>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-steps-item-description"
                  >
                    <div
                      class="ant-space ant-space-vertical full-width-element"
                    >
                      <div
                        class="ant-space-item"
                        style="margin-bottom: 0px;"
                      >
                        <span>
                          workflowTemplate.form.label.execUser
                           : 
                          workflowTemplate.progressConfig.exec.executeUserType.matchExecute
                        </span>
                      </div>
                      <div
                        class="ant-space-item"
                      >
                        <span>
                          <span
                            class="text-black"
                          >
                            workflowTemplate.form.label.reviewDesc
                             : 
                          </span>
                          3
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
